pkg/services/ngalert/api/api_ruler.go

```diff
@@ -218,36 +219,43 @@ func (srv RulerSrv) RoutePostNameRulesConfig(c *models.ReqContext, ruleGroupConf
 		return toNamespaceErrorResponse(err)
 	}
 
-	// quotas are checked in advanced
-	// that is acceptable under the assumption that there will be only one alert rule under the rule group
-	// alternatively we should check the quotas after the rule group update
-	// and rollback the transaction in case of violation
-	limitReached, err := srv.QuotaService.QuotaReached(c, "alert_rule")
-	if err != nil {
-		return ErrResp(http.StatusInternalServerError, err, "failed to get quota")
-	}
-	if limitReached {
-		return ErrResp(http.StatusForbidden, errors.New("quota reached"), "")
-	}
-
-	// TODO validate UID uniqueness in the payload
-
 	//TODO: Should this belong in alerting-api?
 	if ruleGroupConfig.Name == "" {
 		return ErrResp(http.StatusBadRequest, errors.New("rule group name is not valid"), "")
 	}
 
-	var alertRuleUIDs []string
+	alertRuleUIDs := make(map[string]struct{})
 	for _, r := range ruleGroupConfig.Rules {
 		cond := ngmodels.Condition{
 			Condition: r.GrafanaManagedAlert.Condition,
 			OrgID:     c.SignedInUser.OrgId,
 			Data:      r.GrafanaManagedAlert.Data,
 		}
 		if err := validateCondition(cond, c.SignedInUser, c.SkipCache, srv.DatasourceCache); err != nil {
-			return ErrResp(http.StatusBadRequest, err, "failed to validate alert rule %s", r.GrafanaManagedAlert.Title)
+			return ErrResp(http.StatusBadRequest, err, "failed to validate alert rule %q", r.GrafanaManagedAlert.Title)
```

Comment:
Is `%q` correct here?

Reply:
The difference is that the string is double-quoted so the error is more readable in case the rule title contains spaces. not important but I think I prefer it that way.